File: ExtStore2/Models/CartItem.cs
using System.ComponentModel.DataAnnotations;

namespace ExtStore2.Models
{
    public class CartItem
    {
        public int CartItemId { get; set; }
        public string UserId { get; set; }
        public int ProductId { get; set; }
        public int Count { get; set; }

        //public virtual User User { get; set; }
        public virtual Product Product { get; set; }
    }
}

File: ExtStore2/app/model/CartItem.js
Ext.define('Front.model.CartItem', {
    extend: 'Ext.data.Model',
    fields: ['CartItemId', 'ProductId', 'Count'],

    associations: [{
        type: 'hasOne',
        model: 'Front.model.Product',
        primaryKey: 'CartItemId',
        foreignKey: 'ProductId',
        autoLoad: true,
        associationKey: 'Product',
        //name: 'orderitems'
    }],
});